server/src/com/cloud/network/IpAddressManagerImpl.java

```diff
@@ -1394,6 +1380,25 @@ public IPAddressVO associateIPToGuestNetwork(long ipId, long networkId, boolean
         }
     }
 
+    protected boolean isSourceNatAvailableForNetwork(long networkId, Account owner, IPAddressVO ipToAssoc, Network network) {
```

Comment:
is networkId needed on this signature? network is being passed as well!

Reply:
no, not really. must be a result of intellij's extract method